packages/pg/lib/client.js

```diff
@@ -605,17 +604,15 @@ class Client extends EventEmitter {
     return new this._Promise((resolve, reject) => {
       this._connect((error) => {
         if (error) {
-          if (this.connectionParameters.loadBalance) {
+          logger.silly("Not able to connect to " + this.host + " due to error " + error.message)
+          if (this.connectionParameters.loadBalance && Client.hostServerInfo.size !== 0) {
             if (Client.hostServerInfo.has(this.host)) {
               logger.debug("Adding " + this.host + " to failed host list")
               Client.failedHosts.set(this.host, Client.hostServerInfo.get(this.host))
               let start = new Date().getTime();
               Client.failedHostsTime.set(this.host, start)
               Client.connectionMap.delete(this.host)
               Client.hostServerInfo.delete(this.host)
-            } else if (Client.failedHosts.has(this.host)) {
-              Client.failedHosts.delete(this.host)
-              Client.failedHostsTime.delete(this.host)
```

Comment:
Let us postpone this change to later, if it is not proven to be causing a failure.